contrib/lib/src/space_helmet/helmet.rs

```diff
@@ -0,0 +1,287 @@
+/// as shown.
+/// ```rust
+/// # extern crate rocket;
+/// # extern crate rocket_contrib;
+///
+/// use rocket_contrib::space_helmet::SpaceHelmet;
+///
+/// let helmet = SpaceHelmet::default();
+/// let rocket = rocket::ignite().attach(helmet);
+///
+/// ```
+///
+/// To enable an additional header, call the method for that header, with the
+/// policy for that header, before attach e.g.
+/// ```rust
+/// # extern crate rocket;
+/// # extern crate rocket_contrib;
+///  use rocket_contrib::space_helmet::{SpaceHelmet, ReferrerPolicy};
+///  let helmet = SpaceHelmet::default().referrer_policy(ReferrerPolicy::NoReferrer);
+///  let rocket = rocket::ignite().attach(helmet);
+/// ```
+///
+/// To disable a header, call the method for that header with `None` as
+/// the policy.
+/// ```rust
+///
+/// # extern crate rocket;
+/// # extern crate rocket_contrib;
+///  use rocket_contrib::space_helmet::SpaceHelmet;
+///  let helmet = SpaceHelmet::default().no_sniff(None);
+///  let rocket = rocket::ignite().attach(helmet);
+/// ```
+///
+/// `SpaceHelmet` supports the builder pattern to configure multiple policies
+/// ```rust
+/// # extern crate rocket;
+/// # extern crate rocket_contrib;
+///  use rocket_contrib::space_helmet::{HSTSPolicy, ExpectCTPolicy, ReferrerPolicy, SpaceHelmet};
+///  let helmet = SpaceHelmet::default()
+///     .hsts(HSTSPolicy::default())
+///     .expect_ct(ExpectCTPolicy::default())
+///     .referrer_policy(ReferrerPolicy::default());
+///
+///  let rocket = rocket::ignite().attach(helmet);
+/// ```
+///
+/// # TLS and HSTS
+///
+/// If tls is enabled when rocket
+/// [launched()'ed](../../rocket/fairing/trait.Fairing.html) in a
+/// non-development environment e.g.
+/// [Staging]( ../../rocket/struct.Config.html#method.staging)
+/// or [Production]( ../../rocket/struct.Config.html#method.production),
+/// `SpaceHelmet` enables hsts with its default policy and issue a warning.
+/// To get rid of this warning, set an [hsts](#method.hsts) policy if you are
+/// using tls.
+///
+pub struct SpaceHelmet<'a> {
+    expect_ct_policy: Option<ExpectCTPolicy<'a>>,
+    no_sniff_policy: Option<NoSniffPolicy>,
+    xss_protect_policy: Option<XSSPolicy<'a>>,
+    frameguard_policy: Option<FramePolicy<'a>>,
+    hsts_policy: Option<HSTSPolicy>,
+    force_hsts_policy: Option<HSTSPolicy>,
+    force_hsts: AtomicBool,
+    referrer_policy: Option<ReferrerPolicy>,
+}
+
+// helper for Helmet.apply
+macro_rules! try_apply_header {
+    ($self:ident, $response:ident, $policy_name:ident) => {
+        if let Some(ref policy) = $self.$policy_name {
+            if $response.set_header(policy) {
+                warn_!(
+                    "(Space Helmet): set_header failed, found existing header \"{}\"",
+                    Header::from(policy).name
+                );
+            }
+        }
+    };
+}
+
+impl<'a> SpaceHelmet<'a> {
+    /// Returns a new `SpaceHelmet` instance with `no_sniff`, `frameguard`,
```

Comment:
This should probably say that you enable `NoSniffPolicy::default()`, `FramePolicy::default()`, `XSSPolicy::default()` and `HSTSPolicy::default()`, linking to the documentation for the methods, which should be written.